feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -909,8 +966,67 @@ def _create(
         # Handle parameters
         if mapping is None:
             raise ValueError("`mapping` is required for this Protocol")
-        if extends:
-            raise NotImplementedError("Can't extend simulations yet")
+
+        extends_data = {}
+        if isinstance(extends, ProtocolDAGResult):
+
+            if not extends.ok():
+                raise ValueError("Cannot extend protocols that failed")
+
+            setup = extends.protocol_units[0]
+            simulations = extends.protocol_units[1:-1]
+
+            r_setup = extends.protocol_unit_results[0]
+            r_simulations = extends.protocol_unit_results[1:-1]
+
+            # confirm consistency
+            original_state_a = setup.inputs["state_a"].key
+            original_state_b = setup.inputs["state_b"].key
+            original_mapping = setup.inputs["mapping"]
+
+            if original_state_a != stateA.key:
+                raise ValueError(
+                    "'stateA' key is not the same as the key provided by the 'extends' ProtocolDAGResult."
+                )
+
+            if original_state_b != stateB.key:
+                raise ValueError(
+                    "'stateB' key is not the same as the key provided by the 'extends' ProtocolDAGResult."
+                )
+
+            if mapping is not None:
+                if original_mapping != mapping:
+                    raise ValueError(
+                        "'mapping' is not consistent with the mapping provided by the 'extends' ProtocolDAGResult."
+                    )
+
+            # TODO: are there instances where this is too strict?
+            if setup.inputs["settings"] != self.settings:
+                raise ValueError(
+                    "protocol settings are not consistent with those present in the SetupUnit of the 'extends' ProtocolDAGResult."
+                )
```

Comment:
I think this is a good check to have, and is not too strict at all. The philosophy behind a `Protocol` with a given `Settings` object is that those `Settings` are immutable once given to the `Protocol` object, and so calling `Protocol.create(..., extends=<protocol_dag_result>)` should feature identical settings between the calling `Protocol` and the `ProtocolDAGResult` we wish to extend from.

Reply:
Just chiming in here - there are a few cases where this is going to be "not ideal" (mostly QoL but possibly could be frustrating).

So one example here could be a case where you extend but you switch compute platform - the results _should_ be the same, but the settings will be different. Similarly the trajectory write frequency (you can imagine a case where someone decides that it's just not that necessary to have the trajectory after extending), etc...

It's of course not a blocker right now, but a "smarter" equality might be necessary (e.g. https://github.com/OpenFreeEnergy/gufe/issues/329).

Reply:
I see your point, and think it makes sense to have a `thermodynamically_equals` method or function for `Protocol` settings objects for this purpose.

Systems like `alchemiscale` may limit the way users can do `extends` in the sense that `Task`s may only extend other `Task`s from the same `Transformation`, but the `Protocol` system on its own isn't as strict, since the concept of a `Transformation` sits outside of it.